packages/fuels-programs/src/call_utils.rs

```diff
@@ -230,15 +247,38 @@ pub(crate) fn get_single_call_instructions(offsets: &CallOpcodeParamsOffset) ->
         .asset_id_offset
         .try_into()
         .expect("asset_id_offset out of range");
-    let instructions = [
+
+    let mut instructions = [
         op::movi(0x10, call_data_offset),
         op::movi(0x11, gas_forwarded_offset),
         op::lw(0x11, 0x11, 0),
         op::movi(0x12, amount_offset),
         op::lw(0x12, 0x12, 0),
         op::movi(0x13, asset_id_offset),
         op::call(0x10, 0x12, 0x13, 0x11),
-    ];
+    ]
+    .to_vec();
+    // The instructions are different if you want to return data that was on the heap
+    if let ParamType::Vector(inner_param_type) = output_param_type {
+        let inner_type_byte_size: u16 =
+            (inner_param_type.compute_encoding_width() * WORD_SIZE) as u16;
+        instructions.extend([
```

Comment:
Also a potential candidate for a helper method.

Reply:
I don't really like yet another helper method inside an helper method.